### ISSUE TITLE
Breach alert system

### ISSUE THREAD

**Issue by webknjaz** [patron-it/opencanary#3]
* [x] Sending notifications to the central server
* [x] URL stored in config
* [x] Event caching/buffering
* [x] Decide on secret placement

**Comment by martin-haller** [comment]
I have tested the new logging module. It is sending the HTTP Posts correctly. Only I am missing the "secret". Currently I have it in a config on a root level as "logger.secret". 

If you want, it could be also configured under the logger configuratin (so it would make more sense as it is not a "global" property).

```json
{                                                                                                    
   "device.node_id": "opencanary-1",
   "logger": {
        "class": "PyLogger",
        "kwargs": {
            "formatters": {
                "plain": {
                    "format": "%(message)s"
                }
            },
            "handlers": {
                "console": {
                    "class": "logging.StreamHandler",
                    "stream": "ext://sys.stdout"
                },
                "file": {
                    "class": "logging.FileHandler",
                    "filename": "/var/log/opencanary.log"
                },
                "HTTPAlert": {
                    "class": "opencanary.logger.HTTPAlertHandler",
                    "server_url": "http://url",
                    "secret": "ABCDEF123456789"
                }
            }
        }
    }
}
```

Current HTTP POST look as captured by Wireshark:
![image](https://user-images.githubusercontent.com/34276685/47349747-9100b480-d6b4-11e8-8787-80f071e5219a.png)

**Comment by webknjaz** [comment]
Ok, I'll add it. I think, I'll send it as a [Bearer token](https://swagger.io/docs/specification/authentication/bearer-authentication/) ([RFC 6750](https://tools.ietf.org/html/rfc6750)), like:
```http
Authorization: Bearer <token>
```

**Comment by webknjaz** [comment]
So now you should be able to do like this:
```diff
 {
    "device.node_id": "opencanary-666",
    "logger": {
         "class": "PyLogger",
         "kwargs": {
             "formatters": {
                 "plain": {
                     "format": "%(message)s"
                 }
             },
             "handlers": {
                 "console": {
                     "class": "logging.StreamHandler",
                     "stream": "ext://sys.stdout"
                 },
                 "file": {
                     "class": "logging.FileHandler",
                     "filename": "/var/log/opencanary.log"
                 },
                 "HTTPAlert": {
                     "class": "opencanary.logger.HTTPAlertHandler",
                     "server_url": "http://url",
-                    "secret": "ABCDEF123456789"
+                    "client_id": 666,
+                    "secret": "abcdefghijklmnopqrstuvwxyz0123456789"
                 }
             }
         }
     }
 }
```

I've also added `client_id`, so that you could identify the device in logs. The header for the config above would look like:
```http
Authorization: Bearer 666-abcdefghijklmnopqrstuvwxyz0123456789
```

**Comment by martin-haller** [comment]
Thx. I will try it on Tuesday or Wednesday and let you know.

**Comment by martin-haller** [comment]
After the update I am unable to launch the Opencanaryd. I get the following error:

```
root@Diskstation-4:/etc/opencanaryd# opencanaryd
[-] Using config file: opencanary.conf
Traceback (most recent call last):
  File "/home/opencanary/env/bin/opencanaryd", line 11, in <module>
    load_entry_point('patron-it-opencanary==0.5.0.post1.dev1+gc6b1554', 'console_scripts', 'opencanaryd')()
  File "build/bdist.linux-armv7l/egg/opencanary/cli.py", line 126, in main
  File "/home/opencanary/env/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/opencanary/env/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/opencanary/env/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/opencanary/env/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/opencanary/env/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "build/bdist.linux-armv7l/egg/opencanary/cli.py", line 122, in cli
TypeError: 'NoneType' object is not callable
root@Diskstation-4:/etc/opencanaryd#
```

**Comment by webknjaz** [comment]
Oh, I didn't handle "no args" case. `--dev` worked, but I didn't check this one. I'll fix it :)

**Comment by webknjaz** [comment]
Fixed with 0cca75f

**Comment by webknjaz** [comment]
@Patrhal I've added an exponential backoff to sending individual log messages, I hope it'll do for buffering.
